src/app/_components/CheckBox/CheckBox.js

```diff
@@ -0,0 +1,172 @@
+'use client';
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import cn from '@/utils/cn';
+import { checkBoxVariants } from './styles';
+
+export default function CheckBox({
+  value,
+  onChange,
+  name,
+  text = '',
+  subText = '',
+  disabled = false,
+  checked = false,
+  children = null,
+  variant = checkBoxVariants.default,
+}) {
+  const id = `checkbox_id_${name}_|_${value}__`;
+  return (
+    <div className={cn(variant.container.base)}>
+      <input
+        id={id}
+        name={name}
+        className={cn(
+          variant.checkBox.base,
+          variant.checkBox.hover,
+          variant.checkBox.focus,
+          variant.checkBox.focusRing,
+          variant.checkBox.checked,
+          variant.checkBox.checkedFocus,
+          variant.checkBox.checkedHover,
+          variant.checkBox.disabled,
+        )}
+        value={value}
+        onChange={onChange}
+        type="checkbox"
+        checked={checked}
+        disabled={disabled}
+      />
+      <label
+        className={cn(
+          variant.labelBefore.base,
+          variant.labelBefore.layout,
+          variant.labelBefore.peerFocus,
+          variant.labelBefore.peerChecked,
+          variant.labelBefore.peerDisabled,
+        )}
+        htmlFor={id}
+      >
+        {checked && (
+          <Tick
+            styles={cn(
```

Comment:
@IvanKoshan Consider renaming `styles` prop to `className`. To make clear that it is not for inline styles.

Reply:
did it

Reply:
Don't forget to rename prop on line 53 when you use imported <TickIcon/>